src/data/providers/balancer-api/client/index.ts

```diff
@@ -3,9 +3,15 @@ import { ChainId } from '../../../../utils';
 export class BalancerApiClient {
     apiUrl: string;
     chainId: ChainId;
+    clientName: string;
+    clientVersion: string;
+
     constructor(apiUrl: string, chainId: ChainId) {
         this.apiUrl = apiUrl;
         this.chainId = chainId;
+
+        this.clientName = process.env.API_CLIENT_NAME || 'balancer-sdk';
+        this.clientVersion = process.env.API_CLIENT_VERSION || __PACKAGE_VERSION__;
```

Comment:
Test is failing because __PACKAGE_VERSION__ is not defined during tests (only during build).
I see that you're assigning JSON.stringify(pkg.version) to it on the build config.
Why don't you simply default the clientVersion here to that, instead of assigning it on build config only?
Here's my suggestion:
```suggestion
        this.clientVersion = process.env.API_CLIENT_VERSION || JSON.stringify(pkg.version);
```